### ISSUE TITLE
A005 and A002 both trigger for lambda's

### ISSUE THREAD

**Issue by jaap3** [astral-sh/ruff#14135]
This example triggers two violations (with preview enabled), is this intentional?

```python
lambda id: id + 1
```

```
$ ruff check --isolated --preview --select A test.py

test.py:1:8: A002 Argument `id` is shadowing a Python builtin
  |
1 | lambda id: id + 1
  |        ^^ A002
  |

test.py:1:8: A006 Lambda argument `id` is shadowing a Python builtin
  |
1 | lambda id: id + 1
  |        ^^ A006
  |
```

```
$ ruff --version
ruff 0.7.2
```

**Comment by dylwil3** [comment]
Nice catch! It looks like `flake8-builtins` distinguishes between (named) functions and lambdas. We should match that behavior and maybe update the documentation on `A002` and `A006`.

My only hesitation is that this would be a breaking change for a stable rule - so if people were running the checker with preview _off_ and catching this shadowing on lambdas, they would no longer catch it if we matched `flake8-builtins`. 

Not sure the best protocol for handling this - @MichaReiser / @AlexWaygood ?

**Comment by AlexWaygood** [comment]
Great question. When we _expand_ the scope of a stable rule, we generally put the new behaviour in preview mode for a couple of months before promoting it to stable. When we _reduce_ the scope of a rule, it's more of a judgement call.

I _can_ see the argument that this might be a breaking change for some people -- but it definitely feels like a bug to me to have two rules complain about the exact same error. So I'd be inclined to ship the behaviour change straight into stable for this one, so that users can benefit from the bugfix even if they haven't selected `preview = true` (most don't).
